draft-ietf-oauth-attestation-based-client-auth.md

```diff
@@ -118,83 +120,40 @@ Client Instance:
 
 ## Client Attestation JWT {#client-attestation-jwt}
 
-The following rules apply to validating the Client Attestation JWT. Application of additional restrictions and policy are at the discretion of the Authorization Server.
+The Client Attestation MUST be encoded as a "JSON Web Token (JWT)" according to {{RFC7519}}.
 
-1. The JWT MUST contain an "iss" (issuer) claim that contains a unique identifier for the entity that issued the JWT. In the absence of an application profile specifying otherwise, compliant applications MUST compare issuer values using the Simple String Comparison method defined in Section 6.2.1 of {{RFC3986}}.
+The following content applies to the JWT Header:
 
-2. The JWT MUST contain a "sub" (subject) claim with a value corresponding to the "client_id" of the OAuth client.
+* `typ`: REQUIRED. The JWT type MUST be `oauth-client-attestation+jwt`.
 
-3. The JWT MUST contain an "exp" (expiration time) claim that limits the time window during which the JWT can be used.  The authorization server MUST reject any JWT with an expiration time that has passed, subject to allowable clock skew between systems.
+The following content applies to the JWT Claims Set:
 
-4. The JWT MUST contain an "cnf" claim conforming {{RFC7800}} that conveys the key to be used for producing the client attestation pop for client authentication with an authorization server. The key MUST be expressed using the "jwk" representation.
+* `iss`: REQUIRED. The `iss` (subject) claim MUST contains a unique identifier for the entity that issued the JWT. In the absence of an application profile specifying otherwise, compliant applications MUST compare issuer values using the Simple String Comparison method defined in Section 6.2.1 of {{RFC3986}}.
+* `sub`: REQUIRED. The `sub` (subject) claim MUST specify client_id value of the OAuth Client.
+* `exp`: REQUIRED. The `exp` (expiration time) claim MUST specify the time at which the Client Attestation is considered expired by its issuer. The authorization server MUST reject any JWT with an expiration time that has passed, subject to allowable clock skew between systems.
+* `cnf`: REQUIRED. The `cnf` (confirmation) claim MUST specify a key conforming to {{RFC7800}} that is used by the Client Instance to generate the Client Attestation PoP JWT for client authentication with an authorization server. The key MUST be expressed using the "jwk" representation.
+* `iat`: OPTIONAL. The `iat` (issued at) claim MUST specify the time at which the Client Attestation was issued.
+* `nbf`: OPTIONAL. The `nbf` (not before) claim MUST specify the time before which the Client Attestation MUST NOT be accepted for processing.
 
-5. The JWT MAY contain an "nbf" (not before) claim that identifies the time before which the token MUST NOT be accepted for processing.
+The following additional rules apply:
 
-6. The JWT MAY contain an "iat" (issued at) claim that identifies the time at which the JWT was issued.
+1. The JWT MAY contain other claims.
```

Comment:
Should we explain what happens if claims are not understood by a RP?
We have 2 options and should probably clearly state how to deal with such a situation in the spec:
- ignore not understood claims
- fail validation of the client attestation

Reply:
```suggestion
1. The JWT MAY contain other claims. All claims that are not understood by implementations MUST be ignored.
```